### PR TITLE
KP-761 [Ansible] flatten, eslint, dockerfy, reicon Ansible plugin

### PR DESCRIPTION
**Pull request by xkcm**
[**KP-761 in Jira**](https://kaholo.atlassian.net/browse/KP-761)

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          "name": "sshUsername",
+          "viewName": "SSH Username",
+          "description": "Username under which the SSH connection will be obtained.",
+          "placeholder": "ubuntu",
+          "type": "string"
+        },
+        {
+          "name": "sshPass",
+          "viewName": "SSH Password",
+          "description": "Password for the provided username",
```
Comment (graddaniel) [inline config.json:35]: missing dot

File: helpers.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  sshKeyPath,
+  playbookPath,
+  inventoryFiles,
+  inventoryIPs: inventoryIps,
+  limit,
+  modules,
+  vars,
+}) {
+  const sshCredentials = {};
+  if (sshUsername?.trim()) {
```
Comment (graddaniel) [inline helpers.js:17]: With introduction of this change, trimming no longer will be necessary

File: helpers.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+function parseVarsParam(varsParam) {
+  if (_.isNil(varsParam)) {
+    return {};
+  }
+
+  if (_.isPlainObject(varsParam)) {
+    return varsParam;
+  }
+
+  if (_.isArray(varsParam)) {
```
Comment (graddaniel) [inline helpers.js:47]: Please reverse the logic to perform the parsing (happy path) at the end of the function block and return early by throwing the error (unhappy path) when the param doesn't meet our expectations.

**Review by graddaniel — CHANGES_REQUESTED** [review]

File: ansible-cli.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    throw new Error(error.stdout || error.stderr || error.message || error);
+  }
+
+  return result;
+}
+
+async function parseAnsibleConfig(rawAnsibleConfig) {
+  let environmentVariables = {};
+  const volumeConfigs = [];
+  const ansibleConfig = rawAnsibleConfig;
```
Comment (graddaniel) [inline ansible-cli.js:49]: In whole this function you're mixing `ansibleConfig` and `rawAnsibleConfig`. Pick which is the information source and which is the result and then use former to read the data and latter to save it and return it.

File: app.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const kaholoPluginLibrary = require("kaholo-plugin-library");
+const { execute } = require("./ansible-cli");
+const { prepareRunPlaybookPayload } = require("./helpers");
+
+function runPlaybook(params) {
+  const payload = prepareRunPlaybookPayload(params);
```
Comment (graddaniel) [inline app.js:6]: `payload` contains pretty much the same data as `params` witch such difference that some of them are parsed or trimmed. Does that change the nature of the data and makes them no longer params?

File: app.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+  return execute({
+    command: "ansible-playbook",
+    config: payload,
```
Comment (graddaniel) [inline app.js:10]: considering above, the `execute` function accepts some kind of `params` not a config.

File: ansible-cli.js [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+async function execute({
+  config: unparsedAnsibleConfig,
+  command,
+}) {
+  const {
+    volumeConfigs,
+    environmentVariables,
+    ansibleConfig,
+  } = await parseAnsibleConfig(unparsedAnsibleConfig);
```
Comment (graddaniel) [inline ansible-cli.js:25]: A function should return only one result, but let's keep that for the second round of review.

Considering that the `unparsedAnsibleConfig` are actually some kind of params, it makes much more sense to produce config from it, what's pretty much reflected in this function name.

**Review by graddaniel — CHANGES_REQUESTED** [review]
more comments

File: helpers.js [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+ * element in the passed list
+ */
+function prependNamedArguments(rawArguments, namedArgument) {
+  return rawArguments.map((arg) => [namedArgument, arg]).flat();
+}
+
+function extractMountPointsFromVolumeConfigs(volumeConfigs) {
+  return volumeConfigs
+    .map((volumeConfig) => volumeConfig.mountPoint)
+    .map((modulePath) => `$${modulePath}`);
```
[the remark this answers is not in the document]
Comment (xkcm) [reply helpers.js:70]: Yes, I am extracting the mounting points from volume configs and adding a $ before so they are treated as environment variables in shell.

**Review by graddaniel — APPROVED** [review]

**Pull request merged**